src/test/java/com/klodnicki/Bike/v2/service/ChargingStationServiceHandlerTest.java

```diff
@@ -0,0 +1,158 @@
+        Long actual = chargingStationRepository.count();
+
+        //then
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void add_ShouldReturnStationForAdminResponseDTO_WhenGivenChargingStation() {
+        //given
+        chargingStationRepository.deleteAll();
+        ChargingStation chargingStation = new ChargingStation();
+        StationForAdminResponseDTO expected = modelMapper.map(chargingStation, StationForAdminResponseDTO.class);
+
+        //when
+        StationForAdminResponseDTO actual = chargingStationServiceHandler.add(chargingStation);
+
+        //then
+        assertNotNull(actual.getId());
+        expected.setId(actual.getId());
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void findAll_ShouldReturnListOfStationForAdminResponseDTO_WhenExistInDatabase() {
+        //given
+        chargingStationRepository.deleteAll();
+        List<StationForAdminResponseDTO> expectedStationDTOS = new ArrayList<>();
+        for (int i = 0; i < 5; i++) {
+            ChargingStation chargingStation = new ChargingStation();
+            chargingStationRepository.save(chargingStation);
+            StationForAdminResponseDTO stationDTO = modelMapper.map(chargingStation, StationForAdminResponseDTO.class);
+            expectedStationDTOS.add(stationDTO);
+        }
+
+        //when
+        List<StationForAdminResponseDTO> actual = chargingStationServiceHandler.findAll();
+
+        //then
+        assertEquals(expectedStationDTOS, actual);
+    }
+
+    @Test
+    public void findById_ShouldReturnStationForAdminResponseDTO_WhenExistInDatabase() {
+        //given
+        chargingStationRepository.deleteAll();
+        ChargingStation chargingStation = new ChargingStation();
+        chargingStationRepository.save(chargingStation);
+        StationForAdminResponseDTO expected = modelMapper.map(chargingStation, StationForAdminResponseDTO.class);
+
+        //when
+        StationForAdminResponseDTO actual = chargingStationServiceHandler.findById(chargingStation.getId());
+
+        //then
+        assertEquals(expected, actual);
+    }
+
+
+    @Test
+    public void findStationById_ShouldReturnChargingStation_WhenExistInDatabase() {
+        //given
+        chargingStationRepository.deleteAll();
+        ChargingStation chargingStation = new ChargingStation();
+        chargingStationRepository.save(chargingStation);
+
+        //when
+        ChargingStation actual = chargingStationServiceHandler.findStationById(chargingStation.getId());
+        actual.setBikeList(null);
+
+        //then
+        assertEquals(chargingStation, actual);
+    }
+
+    @Test
+    public void findStationById_ShouldThrowIllegalArgumentException_WhenNotFoundChargingStationInDatabase() {
+        //given
+        chargingStationRepository.deleteAll();
+
+        //when
+        //then
+        assertThrows(IllegalArgumentException.class, () -> chargingStationServiceHandler.findStationById(1L));
+    }
```

Comment:
As this is right now being tested. IllegalArgumentException should be replaced with custom exception. This one is too generic.

Reply:
yes sir!